social-middleware/src/application/application.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { InjectPinoLogger, PinoLogger } from 'nestjs-pino';
+import { v4 as uuidv4 } from 'uuid';
+import { Model } from 'mongoose';
+import { Application, ApplicationDocument } from './schemas/application.schema';
+import { FormParameters, FormParametersDocument } from './schemas/form-parameters.schema';
+import { CreateApplicationDto } from './dto/create-application.dto';
+
+@Injectable()
+export class ApplicationService {
+    constructor(
+        @InjectModel(Application.name)
+        private applicationModel: Model<ApplicationDocument>,
+        @InjectModel(FormParameters.name)
+        private formParametersModel: Model<FormParametersDocument>,
+        @InjectPinoLogger(ApplicationService.name)
+        private readonly logger: PinoLogger,
+    ) { }
+
+    async createApplication(dto: CreateApplicationDto): Promise<{ formAccessToken: string }> {
+        const applicationId = uuidv4();
```

Comment:
this service should have error handling.. should be a try { } catch (error) {
log error
throw new InternalServiceError('Failed to create application');